src/metatensor/models/utils/logging.py

```diff
@@ -0,0 +1,156 @@
+        epoch: int,
+        train_loss: float,
+        validation_loss: float,
+        train_info: Dict[str, float],
+        validation_info: Dict[str, float],
+    ):
+        """
+        Log the training metrics.
+
+        The training metrics are automatically aligned to make them easier to read,
+        based on the order of magnitude of each metric at the start of the training.
+
+        Args:
+            epoch: The current epoch.
+            train_loss: The current training loss.
+            validation_loss: The current validation loss.
+            train_info: The current training metrics.
+            validation_info: The current validation metrics.
+        """
+
+        # The epoch is printed with 4 digits, assuming that the training
+        # will not last more than 9999 epochs
+        logging_string = (
+            f"Epoch {epoch:4}, train loss: "
+            f"{train_loss:{self.digits['train_loss'][0]}.{self.digits['train_loss'][1]}f}, "  # noqa: E501
+            f"validation loss: "
+            f"{validation_loss:{self.digits['validation_loss'][0]}.{self.digits['validation_loss'][1]}f}"  # noqa: E501
+        )
+        for name, information_holder in zip(
+            ["train", "valid"], [train_info, validation_info]
+        ):
+            for key, value in information_holder.items():
+                new_key = key
+                if key.endswith("_positions_gradients"):
+                    # check if this is a force
+                    target_name = key[: -len("_positions_gradients")]
+                    if (
+                        self.model_capabilities.outputs[target_name].quantity
+                        == "energy"
+                    ):
+                        # if this is a force, replace the ugly name with "force"
+                        if self.only_one_energy:
+                            new_key = "force"
+                        else:
+                            new_key = f"force[{target_name}]"
+                elif key.endswith("_displacement_gradients"):
+                    # check if this is a virial/stress
+                    target_name = key[: -len("_displacement_gradients")]
+                    if (
+                        self.model_capabilities.outputs[target_name].quantity
+                        == "energy"
+                    ):
+                        # if this is a virial/stress,
+                        # replace the ugly name with "virial/stress"
+                        if self.only_one_energy:
+                            new_key = "virial/stress"
+                        else:
+                            new_key = f"virial/stress[{target_name}]"
+                logging_string += (
+                    f", {name} {new_key} RMSE: "
+                    f"{value:{self.digits[f'{name}_{key}'][0]}.{self.digits[f'{name}_{key}'][1]}f}"  # noqa: E501
+                )
+        logger.info(logging_string)
+
+
+def _get_digits(value: float) -> Tuple[int, int]:
```

Comment:
Hmmm I don't think I can use it to do what I want... Yes, what I do is quite cumbersome but I think it's the best solution for now. The function essentially gets a number and decides what parts of the number should be printed so that there are always at least 5 significant digits

Reply:
What's wrong with %12.5e or equivalent?

Reply:
I have the strong feeling that many of the people we're dealing with won't be comfortable with exponential notation

Reply:
Yes, I would go for format specifiers: `f"something {some_float:12.5e}"`. See documentation here: https://docs.python.org/3.11/library/string.html#formatspec

Reply:
> I have the strong feeling that many of the people we're dealing with won't be comfortable with exponential notation

I'm not sure about this, but we can also use `g` specifier for this.

Reply:
I agree. Maybe the `g` is the better option then `f`